src/main/java/org/folio/auth/authtokenmodule/MainVerticle.java

```diff
@@ -629,16 +645,36 @@ we generate a custom token (since we have that power) that
     logger.debug("Getting user permissions for " + username + " (userId " +
             userId + ")");
     long startTime = System.currentTimeMillis();
-    Future<PermissionData> retrievedPermissionsFuture = usePermissionsSource
-      .getUserAndExpandedPermissions(userId, tenant, okapiUrl, permissionsRequestToken,
-        requestId, extraPermissions);
+
+    // Need to check if the user is still active
+    Future<PermissionData> retrievedPermissionsFuture;
+    if (finalUserId != null && !finalUserId.isEmpty()) {
+      Future<Boolean> activeUser = userService.isActiveUser(finalUserId, tenant, okapiUrl, userRequestToken, requestId);
+      retrievedPermissionsFuture = activeUser.compose(b -> {
+        if (b != null && b.booleanValue()) {
+          return usePermissionsSource.getUserAndExpandedPermissions(finalUserId, tenant, okapiUrl,
+              permissionsRequestToken, requestId, extraPermissions);
+        } else {
+          endText(ctx, 401, "Invalid token: user with id " + finalUserId + " is not active");
+          return Future.succeededFuture();
```

Comment:
Shoudln't this be Future.failedFuture() ?

Reply:
That makes sense. Fixed.